test/sol6/reserves/kyberFprReserveV2.js

```diff
@@ -0,0 +1,3422 @@
+        false,
+        tokenInd,
+        amountToSell,
+        user1,
+        false
+      );
+    }
+
+    let testSuites = [
+      "#Test eth and token in reserve",
+      "#Test eth in reserve, token in wallet",
+      "#Test using weth"
+    ]
+    let isUsingWeth = [false, false, true];
+    let isUsingWallet = [false, true, true];
+    let isDoRateValidation = [true, false];
+
+    for(let r = 0; r <= 1; r++) {
+      for(let t = 0; t < testSuites.length; t++) {
+        describe(`${testSuites[t]}, rate validation: ${isDoRateValidation[r]}`, async() => {
+          it("Test few buys with steps", async() => {
+            //init conversion rate
+            await setupConversionRateV2sContractWithStepsV1();
+            await generalSetupReserveContract(isUsingWallet[t], isUsingWeth[t]);
+            await reserveInst.setDoRateValidation(isDoRateValidation[r], {from: admin});
+
+            if (isUsingWeth[t]) {
+              await reserveInst.setTokenWallet(weth.address, walletForToken, {from: admin});
+              await weth.approve(reserveInst.address, MAX_ALLOWANCE, {from: walletForToken});
+            }
+
+            if (isUsingWallet[t]) {
+              // approve
+              for(let i = 0; i < numTokens; i++) {
+                await reserveInst.setTokenWallet(tokenAdd[i], walletForToken, {from: admin});
+                await reserveInst.approveWithdrawAddress(tokenAdd[i], withdrawAddress, true, {from: admin});
+                await tokens[i].approve(reserveInst.address, MAX_ALLOWANCE, {from: walletForToken});
+              }
+            }
+
+            let tokenInd = 2;
+            // make a simple buy and sell to init some data first
+            await makeSimpleTradesV1(tokenInd, isUsingWallet[t], isUsingWeth[t]);
+
+            let gasUsedPerStep = [];
+            let numberTxsPerStep = [];
+            for(let i = 0; i <= 10; i++) {
+              gasUsedPerStep.push(new BN(0));
+              numberTxsPerStep.push(0);
+            }
+            for(let i = 0; i <= 5; i++) {
+              let amountWei = new BN(1000);
+              let data = await tradeAndVerifyDataWithStepsV1(
+                reserveInst,
+                true,
+                tokenInd,
+                amountWei,
+                user1,
+                false
+              );
+              numberTxsPerStep[data.numberSteps]++;
+              gasUsedPerStep[data.numberSteps].iadd(new BN(data.gasUsed));
+            }
+
+            for(let i = 0; i <= 10; i++) {
+              if (numberTxsPerStep[i] > 0) {
+                console.log(`         Average gas used for buy with ${i} steps: ${Math.floor(1.0 * gasUsedPerStep[i]/numberTxsPerStep[i])}`)
+              }
+            }
+          });
+
+          it("Test few sells with steps", async() => {
+            //init conversion rate
+            await setupConversionRateV2sContractWithStepsV1();
+            await generalSetupReserveContract(isUsingWallet[t], isUsingWeth[t]);
+
```

Comment:
better add this setup for all sell runs

await reserveInst.setDoRateValidation(isDoRateValidation[r], {from: admin});


I think its not set well.
and also important if someone running a separate test.

Reply:
notice its also missing in other sell steps loop.

Reply:
Updated, it was missing in sell loops